### PR TITLE
Automatically scrape qualifiers from constraint bodies

### PR DESCRIPTION
**Pull request by ranjitjhala**
Depends on https://github.com/ucsd-progsys/liquid-fixpoint/pull/651

### CODE REVIEW

**Comment by ranjitjhala** [comment]
occurs to me that the CI will probably fail until the fixpoint PR is merged into develop?

File: flux-tests/tests/neg/surface/scrape00.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#![feature(register_tool)]
+#![register_tool(flux)]
```
Comment (nilehmann) [inline flux-tests/tests/neg/surface/scrape00.rs:2]: I'd add an explicit `#![flux::cfg(scrape_quals = false)]` here

File: flux-tests/tests/pos/surface/scrape00.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#![feature(register_tool)]
+#![register_tool(flux)]
+#![feature(custom_inner_attributes)]
+#![flux::cfg(scrape_quals = "true")]
```
Comment (nilehmann) [inline flux-tests/tests/pos/surface/scrape00.rs:4]: I think we can write it without the quotes

```suggestion
#![flux::cfg(scrape_quals = true)]
```

File: flux-fixpoint/src/lib.rs [hunk 3]
```diff
@@ -156,6 +158,9 @@ impl KVar {
 
 impl<Tag: fmt::Display> fmt::Display for Task<Tag> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.scrape_quals {
+            writeln!(f, "(fixpoint \"--scrape=both\")")?;
```
Comment (nilehmann) [inline flux-fixpoint/src/lib.rs:162]: Is `both` necessary to verify the tests we have?

**Review by nilehmann — APPROVED** [review]
LGTM

File: flux-fixpoint/src/lib.rs (hunk 3 above)
Comment (ranjitjhala) [reply flux-fixpoint/src/lib.rs:162]: Good point -- not really. `--both` verifies these examples if you remove ALL the default qualifiers, but `--head` suffices with the default qualifers (as we need something like `x <= y`)

Comment (ranjitjhala) [reply flux-fixpoint/src/lib.rs:162]: So for now, I'd like to leave `both` as default. Another option is to make `scrape_quals` not a `bool` but something like `No|Head|Both` ?

**Pull request merged**